cpp/arcticdb/processing/clause.cpp

```diff
@@ -100,6 +100,7 @@ std::vector<EntityId> PassthroughClause::process(std::vector<EntityId>&& entity_
 }
 
 std::vector<EntityId> FilterClause::process(std::vector<EntityId>&& entity_ids) const {
+    ARCTICDB_DEBUG_THROW(5)
```

Comment:
Could just stick this in the base class to avoid the duplication
https://github.com/man-group/ArcticDB/blob/master/cpp/arcticdb/processing/clause.hpp#L59-L61
And could put one in the `structure_for_processing` overload that takes `entity_ids_vec` as well for good measure.